### PR TITLE
Issue #27: Amigo Request Creation

### PR DESCRIPTION
**Pull request by mckoon**
Issue #27: Amigo Request Creation

Creates AmigoRequestCreateComponent (FAB).
Adds this FAB to Amigos page.
Adds this FAB to Amigo Requests page.
Creates NewAmigoRequestPage.

### CODE REVIEW

File: app/amigo-request-create-component/amigo-request-create.component.html [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<div class="amigo-request-create">
+    <button fab fab-bottom fab-right fab-fixed
+            secondary
+            class="amigo-request-create-fab"
+            [navPush]="onClickPage">
+        <ion-icon name="person-add"></ion-icon>
+    </button>
+</div>
```
Comment (mckoon) [inline app/amigo-request-create-component/amigo-request-create.component.html]: FAB button that gets added on bottom of Amigos and Amigos Requests pages.

Clicking it takes user to new amigo form.

File: app/pages/new-amigo-request/new-amigo-request.html [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+                    Invite
+                </button>
+
+            </ion-item>
+
+        </ion-list>
+
+    </form>
+
+</ion-content>
```
Comment (mckoon) [inline app/pages/new-amigo-request/new-amigo-request.html]: newrequestform

Comment (mckoon) [inline app/pages/new-amigo-request/new-amigo-request.html]: For now, no way to share a different amigo with someone else.  We will have to think through how we will do that in the UI.

File: app/theme/app.core.scss [hunk 3]
```diff
@@ -15,3 +15,10 @@
 @import "../next-dose/next-dose";
 @import "../last-dose/last-dose";
 @import "../user-status-card/user-status-card";
+
+/* Global Classes */
+
+/* Make room on bottom of all pages for FABs. */
+ion-page ion-content[padding] scroll-content {
+  padding-bottom: 100px;
+}
```
Comment (mckoon) [inline app/theme/app.core.scss]: This is so every page has some padding at the bottom so any FAB won't cover content that can't be scrolled above the button.

File: app/pages/amigo-requests/amigo-requests.html [hunk 4]
```diff
@@ -7,4 +7,6 @@
     <amigo-share-request-card *ngFor="let amigoShareRequest of amigoShareRequests"
                               [amigoShareRequest]="amigoShareRequest"></amigo-share-request-card>
 
-</ion-content>
+    <amigo-request-fab></amigo-request-fab>
+
+</ion-content>
```
Comment (mckoon) [inline app/pages/amigo-requests/amigo-requests.html]: Example of FAB:

fab

**Comment by geoffreygraddy** [comment]
:+1:

File: app/pages/new-amigo-request/new-amigo-request.html (hunk 2 above)
Comment (jking31cs) [inline app/pages/new-amigo-request/new-amigo-request.html]: First thoughts, when you click on an amigo from the amigo listing page, have a "share" button that brings you to a page like this.  Maybe list the Amigo's name in the title bar as well for clarity: i.e. "Add Amigo Request for Pebbles Flintstone", or something to show who you're sharing as well as the email of the person you're sharing it with.

**Comment by jleopold28** [comment]
👍

**Pull request merged**